### PR TITLE
DOC: Improve reshape\concat

### LINKED ISSUE #46825: DOC: write guide for how to replace append
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

Probably as examples in `concat`?

### Documentation problem

There are some common use-cases for `append`, and it's not obvious to newcomers how to use `concat` instead

### Suggested fix for documentation

Include some examples of replacing common use-cases of `append`, such as https://github.com/pandas-dev/pandas/issues/35407#issuecomment-1092892819

### PR DESCRIPTION
**Pull request by anetakahle**
- [x] closes #46825
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: pandas/core/reshape/concat.py [hunk 1]
```diff
@@ -173,6 +173,7 @@ def concat(
         they are all None in which case a ValueError will be raised.
     axis : {0/'index', 1/'columns'}, default 0
         The axis to concatenate along.
+        If 0, the outcome is identical to :meth:`DataFrame.append`.
```
Comment (MarcoGorelli) [inline pandas/core/reshape/concat.py:176]: `append` is deprecated, we shouldn't mention it here

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Hey @anetakahle , thanks for taking this on

Instead of this example, could you add an example of how to append a single row (e.g. [here](https://github.com/pandas-dev/pandas/issues/35407#issuecomment-1092892819)), which seems to be the most common source of confusion?

File: pandas/core/reshape/concat.py (hunk 1 above)
Comment (anetakahle) [reply pandas/core/reshape/concat.py]: append mention removed

**Comment by anetakahle** [comment]
@MarcoGorelli thanks for a quick review :)
I've updated the example, is the current version better?

File: pandas/core/reshape/concat.py [hunk 2]
```diff
@@ -344,6 +344,22 @@ def concat(
...
+    >>> a = pd.DataFrame({"A": 1, "B": 2}, index=[0])
+    >>> a
+        A   B
+    0   1   2
+    >>> b = pd.DataFrame({"A": 3}, index=[0])
+    >>> b
+        A
+    0   3
+    >>> for rowIndex, row in b.iterrows():
+    >>>     print(pd.concat([a, row.to_frame().T], ignore_index=True))
```
Comment (MarcoGorelli) [inline pandas/core/reshape/concat.py:359]: does it work to make this
```
new_row = pd.Series([3])
```
and then just do
```
pd.concat([a, new_row.to_frame().T], ignore_index=True)
```
?
Iterating over rows is what we want to avoid

Also, let's rename `a` to `df7`

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/core/reshape/concat.py (hunk 2 above)
Comment (anetakahle) [reply pandas/core/reshape/concat.py]: Changed as requested in 90bafc6d3b223ae8791e631dac582d206b922842

Is it ok that the data from the new_row have appeared in a new row and not in the first one as in the previous solution?

I've also read the discussion about the deprecated append and maybe we should add some note into the documentation as well, something like 

> It is not recomended to build DataFrames by adding single rows in a
not loop. Build a list of rows and make a DataFrame in a single concat.

File: pandas/core/reshape/concat.py [hunk 3]
```diff
@@ -344,6 +344,20 @@ def concat(
...
+    >>> df7
+        a   b
+    0   1   2
+    >>> new_row = pd.Series([3])
+    >>> new_row
+    0   3
+    >>> pd.concat([df7, new_row.to_frame().T], ignore_index=True)
+         a    b    0
+    0  1.0  2.0  NaN
+    1  NaN  NaN  3.0
```
Comment (MarcoGorelli) [inline pandas/core/reshape/concat.py:360]: ah apologies, hadn't realised it would look like this

how about
```
new_rows = pd.Series({'a': 3, 'b': 4})
```
so that then the concatenated version looks like
```
   a  b
0  1  2
0  3  4
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Comment by MarcoGorelli** [comment]
> I've also read the discussion about the deprecated append and maybe we should add some note into the documentation as well, something like

Yes, good suggestion, thanks!

File: pandas/core/reshape/concat.py (hunk 3 above)
Comment (anetakahle) [reply pandas/core/reshape/concat.py]: Changed as requested in 7d4a81fd3c10eeab9f38309bc508373329f0ce4b

File: pandas/core/reshape/concat.py [hunk 4]
```diff
@@ -351,13 +351,16 @@ def concat(
...
+    >>> new_rows = pd.Series({'a': 3, 'b': 4})
+    >>> new_rows
+    0   3   4
+    >>> pd.concat([df7, new_rows.to_frame().T], ignore_index=True)
+         a    b
+    0    1    2
+    1    3    4
+
+    (It is not recomended to build DataFrames by adding single rows in a
+not loop. Build a list of rows and make a DataFrame in a single concat.)
```
Comment (MarcoGorelli) [inline pandas/core/reshape/concat.py:363]: can this go in the `notes` section? Kinda like where the current note is for `DataFrame.append` https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.append.html

File: pandas/core/reshape/concat.py [hunk 5]
```diff
@@ -351,13 +351,16 @@ def concat(
...
         a   b
     0   1   2
-    >>> new_row = pd.Series([3])
-    >>> new_row
-    0   3
-    >>> pd.concat([df7, new_row.to_frame().T], ignore_index=True)
-         a    b    0
-    0  1.0  2.0  NaN
-    1  NaN  NaN  3.0
+    >>> new_rows = pd.Series({'a': 3, 'b': 4})
```
Comment (MarcoGorelli) [inline pandas/core/reshape/concat.py:354]: this will appear as a single row in the DataFrame, so `new_row` should be fine as a name

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
getting there 💪

File: pandas/core/reshape/concat.py (hunk 4 above)
Comment (anetakahle) [reply pandas/core/reshape/concat.py]: Moved in 95139122722c3827cf34a6db7862ac0eeffaf727

File: pandas/core/reshape/concat.py (hunk 5 above)
Comment (anetakahle) [reply pandas/core/reshape/concat.py]: Fixed in 95139122722c3827cf34a6db7862ac0eeffaf727

File: pandas/core/reshape/concat.py [hunk 6]
```diff
@@ -226,6 +226,10 @@ def concat(
     pandas objects can be found `here
     <https://pandas.pydata.org/pandas-docs/stable/user_guide/merging.html>`__.
 
+    
+    It is not recommended to build DataFrames by adding single rows in a
+not loop. Build a list of rows and make a DataFrame in a single concat.
```
Comment (MarcoGorelli) [inline pandas/core/reshape/concat.py:231]:
```suggestion
for loop. Build a list of rows and make a DataFrame in a single concat.
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Lots of unrelated changes, this is showing 15 files changed - could you rebase onto `main` and only modify what's necessary?

Also, please run `pre-commit` on the file(s) you've changed (see https://pandas.pydata.org/docs/development/contributing_codebase.html#pre-commit)

**Comment by anetakahle** [comment]
@MarcoGorelli fixes, shows only 1 file changed now as intended

**Comment by anetakahle** [comment]
@MarcoGorelli I also ran the pre-commit locally and it didn't fail. 

But there are still some other fails here on GitHub, but some of them are on the document as a whole, not just my changes (for example in the case of ```Code Checks / Docstring and typing validation (pull_request)``` the test shows to some errors which are not in the code I posted)

**Comment by MarcoGorelli** [comment]
are you sure you committed your latest changes? can you show the output of:
```
pre-commit run --files pandas/core/reshape/concat.py
git remote -v
git fetch origin
git diff origin/doc-concat
```
please?

**Comment by anetakahle** [comment]
@MarcoGorelli 
Sure!

**Here is the output:**
[**pre-commit run --files pandas/core/reshape/concat.py**](https://pastebin.com/DkNKRYPU)
[**git remote -v**](https://pastebin.com/n7xSBfMN)
**git fetch origin** - no output
[**git diff origin/doc-concat**](https://pastebin.pl/view/3aa7e718)

**Comment by MarcoGorelli** [comment]
OK looks like you have lots of local unsaved changes (perhaps from a merge gone not quite right?)

I think the simplest way out would be:
```
git checkout -- .  # discard local changes
git reset --hard origin/doc-concat
pre-commit run --files pandas/core/reshape/concat.py
git commit -m 'lint file'
git push origin doc-concat
```

**Comment by anetakahle** [comment]
@MarcoGorelli 
Oh, I am so sorry I accidentally ran it from the `main` branch, not the `doc-concat` branch so that was the reason why it came out so weird.
Now I already linted and commited the changes properly.

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
😄 no worries - looks like you've added an extra level on identation to the docstring though, could you restore the original level of indentation please?

**Comment by anetakahle** [comment]
@MarcoGorelli 
The extra level on indentation was added by the pre-commit. 
But I think I fixed it. 😄

File: pandas/core/reshape/concat.py [hunk 7]
```diff
@@ -200,7 +200,7 @@ def concat(
 
         .. versionchanged:: 1.0.0
 
-           Changed to not sort by default.
+            Changed to not sort by default.
```
Comment (MarcoGorelli) [inline pandas/core/reshape/concat.py:203]: here it looks like an extra space has crept in - could you remove it please? This line shouldn't change (likewise for many similar lines below)

File: pandas/core/reshape/concat.py [hunk 8]
```diff
@@ -334,16 +338,30 @@ def concat(
...
+    >>> df7
+        a   b
+    0   1   2
+    >>> new_row = pd.Series({'a': 3, 'b': 4})
+    >>> new_row
+    0   3   4
+    >>> pd.concat([df7, new_row.to_frame().T], ignore_index=True)
+            a    b
+    0    1    2
+    1    3    4
```
Comment (MarcoGorelli) [inline pandas/core/reshape/concat.py:364]:
```suggestion
         a    b
    0    1    2
    1    3    4
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Comment by anetakahle** [comment]
@MarcoGorelli 
Resolved. :)

File: pandas/core/reshape/concat.py (hunk 7 above)
Comment (MarcoGorelli) [inline pandas/core/reshape/concat.py:203]: there's still this unrelated whitespace (and others below) - if you click "File changed" it'll show you everything you've changed, and you can just check your pull request is minimal and only changes what needs changing

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Comment by anetakahle** [comment]
@MarcoGorelli 
OMG, the pre-commit has done so much mess in there :D 
I fixed it, I hope it's all ok and I can squash now? :)

**Review by MarcoGorelli — APPROVED** [review]
Nice, thanks!

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Changes look good, but there's still a doctest failure:
```
_________________ [doctest] pandas.core.reshape.concat.concat __________________
350     ValueError: Indexes have overlapping values: ['a']
351 
352     Append a single row to the end of a ``DataFrame`` object.
353 
354     >>> df7 = pd.DataFrame({'a': 1, 'b': 2}, index=[0])
355     >>> df7
356         a   b
357     0   1   2
358     >>> new_row = pd.Series({'a': 3, 'b': 4})
359     >>> new_row
Expected:
        a   3
        b   4
Got:
    a    3
    b    4
    dtype: int64
```

(and no need to squash the commits)

**Comment by anetakahle** [comment]
@Ok, I think I fixed it, 
but the `Code Checks / Docstring and typing validation` test is still failing and I don't know why :/ :D

**Review by MarcoGorelli — APPROVED** [review]
Awesome, thanks!

The other failure looks unrelated, probably needs to be fixed on `main` first - I'll rebase later to make sure

**Comment by anetakahle** [comment]
@MarcoGorelli 
Thank you😁💜☺️

**Comment by michalkahle** [comment]
Congratulations! :wink:

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Apologies, looks like there's still a related failure:

```
Error: /home/runner/work/pandas/pandas/pandas/core/reshape/concat.py:146:GL03:pandas.concat:Double line break found; please use only one blank line to separate sections or paragraphs, and do not leave blank lines at the end of docstrings
```

**Comment by anetakahle** [comment]
@MarcoGorelli
Oh, I didn't notice. 
I changed it. 

Thank you :)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @anetakahle
